Special v3 (#562)
mlr-org/mlr3book: book/special.qmd

x2 = mean(m$lat)
m = map_data("world")
m = m %>% dplyr::filter(long > -15, long < 40, lat > 36, lat <  60)
m$dist = (-sqrt((m$lon - x1)^2 + (m$lat - x2)^2)) * 10e5
if (min(m$dist) < 0) {
  m$dist = m$dist - min(m$dist)
}
m$rand = runif(nrow(m), min(m$dist), max(m$dist))
cou = m %>% group_by(region) %>% summarize(long = mean(long), lat = mean(lat))

p = ggplot(data = m) +
  scale_fill_gradient2(low = "white", mid = "lightpink", high = "black", midpoint = mean(m$dist)) +
  theme_minimal() +
  xlim(-10, 25) + ylim(37, 58)

t = geom_text(aes(label = region, x = long, y = lat), data = cou, size = 2)

p1 = p + geom_map(map = m, aes(long, lat, map_id = region, fill = rand)) + t +
  theme(legend.position = "n")

p2 = p + geom_map(map = m, aes(long, lat, map_id = region, fill = dist)) + t +
  labs(fill = "Cases")

p1 / p2 & patchwork::plot_layout(guides = "collect")
```

Unlike other tasks we have looked at in this chapter, there is no underlying difference to the implemented learners or measures.

> Instead we provide new resampling methods in `r mlr3spatiotempcv` to account for the similarity in the train and test sets during resampling that originates from spatial autocorrelation.

Reviewer: ```suggestion
Instead we provide additional resampling methods in `r mlr3spatiotempcv` to account for the similarity in the train and test sets during resampling that originates from spatiotemporal autocorrelation.
```

The methods are not new, i.e. all of them existed before and we just "use" them. There is no method that has been newly developed in tempcv.

I think adding a disclaimer here regarding the use of the words "spatial autocor." or "spatiotemporal autocor." would be good. I would prefer to use/introduce the acronym STAC (spatiotemporal autocorrelation) and state that it depends on the data and context whether the autocorrelation is spatial, temporal or spatiotemporal.
This way one does not always have to switch between the different terms in each section and it is clear that the focus is on accounting for a combination of all, if existent.

EDIT: I see there is a disclaimer/note coming further down the text. I'd move it up so readers see it earlier and get clarification on the use of the words and different types of autocor.

lrn("classif.rpart")$train(task)$predict(task)
```

However as discussed above, it is best to use the specialized resampling methods to achieve bias-reduced estimates of model performance.

### Spatiotemporal Cross-Validation {#spatiotemp-cv}

Before we look at the spatial resampling methods implemented in `r mlr3spatiotempcv` we will first show what can go wrong if non-spatial resampling methods are used for spatial data.

> Below we benchmark a decision tree on the `r ref("mlr_tasks_ecuador")` task using two different repeated cross-validation resampling methods, the first ("NSpCV" (non-spatial cross-validation)) is a standard resampling method from `r mlr3`, the second ("SpCV" (spatial cross-validation)) is from `r mlr3spatiotempcv` and is optimized for spatial data.

Reviewer: ```suggestion
Below we benchmark a decision tree on the `r ref("mlr_tasks_ecuador")` task using two different repeated cross-validation resampling methods, the first ("NSpCV" (non-spatial cross-validation)) is a non-spatial resampling method from `r mlr3`, the second ("SpCV" (spatial cross-validation)) is from `r mlr3spatiotempcv` and is optimized for spatial data.
```

  scale_x_continuous(breaks = seq(-79.06, -79.08, -0.02))
```

> Now we have seen why spatial resampling matter we can take a look at what is implemented in `r mlr3spatiotempcv`.

Reviewer: ```suggestion
Now we have seen why spatial resampling matters we can take a look at what methods are available in `r mlr3spatiotempcv`.
```

In this chapter, we explored going beyond deterministic regression and classification to see how functions in the `r mlr3` can be used to implement other machine learning tasks.
Cost-sensitive classification extends the 'normal' classification setting by assuming errors associated with false negatives and false positives are unequal.
Running cost-sensitive classification experiments is possible just using features in `r mlr3`.
Survival analysis, available in `r mlr3proba`, can be thought of as a regression problem when the outcome may be censored, which means it may never be observed within a given time frame.
The final task in `r mlr3proba` is density estimation, the unsupervised task concerned with estimating univariate probability distributions.
Using `r mlr3cluster`, you can perform cluster analysis on observations, which involves grouping observations together according to similarities in their variables.

> Finally, with `r mlr3spatial` and `mlr3spatiotempcv`, it is possible to perform spatial analysis to make predictions using co-ordinates as features and to make spatial predictions.

Reviewer: ```suggestion
Finally, with `r mlr3spatial` and `r mlr3spatiotempcv`, it is possible to perform spatial analysis to make predictions using co-ordinates as features and to make spatial predictions.
```
